### ISSUE TITLE
Gamma() family with inverse-link pretty slow, and a lot of warnings

### ISSUE THREAD

**Issue by strengejacke** [glmmTMB/glmmTMB#1089]
``` r
set.seed(123)
warpbreaks$ID <- sample(1:5, nrow(warpbreaks), replace = TRUE)

tictoc::tic()
m1 <- glmmTMB::glmmTMB(breaks ~ wool * tension + (1 | ID), family = Gamma(), data = warpbreaks)
#> Warning in (function (start, objective, gradient = NULL, hessian = NULL, :
#> NA/NaN function evaluation
#> Warning in finalizeTMB(TMBStruc, obj, fit, h, data.tmb.old): Model convergence
#> ... truncated ...
#> problem; function evaluation limit reached without convergence (9). See
#> vignette('troubleshooting'), help('diagnose')
tictoc::toc()
#> 8.75 sec elapsed

tictoc::tic()
m2 <- lme4::glmer(breaks ~ wool * tension + (1 | ID), family = Gamma(), data = warpbreaks)
tictoc::toc()
#> 0.13 sec elapsed
```

<sup>Created on 2024-08-30 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by bbolker** [comment]
**tl;dr** I'm not sure there is a simple solution here &mdash; nor can I say exactly why `glmmTMB` is having a hard time. The primary difference is that `glmer` profiles out the dispersion parameter (estimating it afterward from the penalized weighted residual deviance), while `glmmTMB` fits it by ML. (This is similar to the difference between the way `glm()` would fit a fixed-effect model and how `glmmTMB()` would do it ...).  This means that `glmmTMB` actually gets a slightly better answer (by 0.5 log-likelihood units).  I suspect that the main problem is  a confounding between dispersion and random-effects SD [although the correlation between these parameters at the final ML is only 0.31]. (One could test this hypothesis by comparing the speed of `glm()` to `glmmTMB()` for a fixed-effects model ...)

I also wonder if the performance would be better on a more realistic example (more levels of the grouping variable, better fitting model) ?

Too much exploration [here](https://github.com/glmmTMB/glmmTMB/blob/master/misc/glmmTMB_GH1089.R).
